fontforge/woff.c

```diff
@@ -348,13 +348,45 @@ return( NULL );
     }
 
     if ( sf!=NULL && metaOffset!=0 ) {
-	char *temp = malloc(metaLenCompressed+1);
+	/*
+	* Boundary/integer overflow checks:
+	*
+	* We don't want to actually dereference a null pointer (returned
+	* by asking to allocate too much RAM) and we don't want to allocate
+	* a 0-sized chunk (caused when one of the (metaLenxxx + 1) values overflows).
+	*
+	* We can safely pass sf->woffMetadata as a NULL pointer because
+	* it's never accessed anywhere else without a check for it being
+	* NULL first
+	*/
+	if(metaLenUncompressed == (unsigned int)0xffffffff) {
```

Comment:
No, no. This still isn't right. The left value is still signed. I'd still redefine `metaLenCompressed` and `metaLenUncompressed` to `uint32_t` and cast from the output of `getlong`.